### PR TITLE
eMASS V3.9 Specification Updates

### PR DESCRIPTION
**Pull request by georgedias**
- [x] Make updates provided by eMASS specifications 3.5 to 3.9
 - [x] Add additional endpoints provide by eMASS specifications 3.5 to 3.9

## How to test
Checkout (`gh pr checkout 1426`) or clone (`git clone -b emass_v3_9  git@github.com:mitre/saf.git`) the PR 
### Install dependencies
```
> git install
```

### Build the executables
```
> npm run prepack
```

### Generate the .env variables 
Run the configure command to generate a .env configuration file containing the necessary environmental variables.

For additional information on how too references [Generating an eMASS Configuration file](https://saf-cli.mitre.org/docs/emasser/#generating-an-emass-configuration-file)

The command to invoke the configure utility from the development environment is:
```
> node bin\run emasser configure (use -h for help)
```
***You need to know the appropriate API and user key, certificates (`key.pem and cert.pem`), key cert password, dev env host url.***

### endpoints added (to test)
```
> node bin\run emasser get dashboards artifacts_summary -o 14
> node bin\run emasser get dashboards artifacts_details -o 14
> node bin\run emasser get dashboards sensor_hardware_summary -o 14
> node bin\run emasser get dashboards sensor_hardware_details -o 14
> node bin\run emasser get dashboards ports_protocols_summary -o 14
> node bin\run emasser get dashboards ports_protocols_summary -o 14
> node bin\run emasser get dashboards ports_protocols_details -o 14
> node bin\run emasser get dashboards fisma_inventory_crypto_summary -o 14
> node bin\run emasser get dashboards va_threat_risk_summary -o 14
> node bin\run emasser get dashboards va_threat_source_details -o 14
> node bin\run emasser get dashboards va_threat_architecture_details -o 14
```
**Note** Some endpoints don't return any data, but should return a 200 status

### Test all emasser endpoints
To run all of the emasser unit tests (51) use this command:
```
> ts-mocha --timeout 8500 --forbid-only .\test\commands\emasser\* 
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,9 +7,9 @@
   "bugs": "https://github.com/mitre/saf/issues",
   "dependencies": {
     "@aws-sdk/client-config-service": "^3.53.0",
-    "@mitre/emass_client": "^3.4.1",
-    "@mitre/hdf-converters": "^2.6.49",
-    "@mitre/heimdall-lite": "^2.6.49",
+    "@mitre/emass_client": "^3",
+    "@mitre/hdf-converters": "^2.6.43",
```
[the remark this answers is not in the document]
Comment (georgedias) [reply package.json:11]: Absolutely, this one slipped by

File: src/commands/emasser/get/artifacts.ts [hunk 2]
```diff
@@ -13,7 +13,7 @@ import {FlagOptions,
 const endpoint = 'artifacts'
 
 export default class EmasserGetArtifacts extends Command {
-  static usage = '<%= command.id %> [ARGUMENTS]';
+  static usage = '<%= command.id %> [ARGUMENT] \n \x1B[93m NOTE: see EXAMPLES for argument case format\x1B[0m';
```
Comment (em-c-rod) [inline src/commands/emasser/get/artifacts.ts:16]: Which command makes this show up?

File: src/commands/emasser/delete/poams.ts [hunk 3]
```diff
@@ -9,7 +9,7 @@ import {outputError} from '../../../utils/emasser/outputError'
 import {FlagOptions, getFlagsForEndpoint} from '../../../utils/emasser/utilities'
 
 export default class EmasserDeletePoams extends Command {
-  static usage = '<%= command.id %> [ARGUMENTS]';
+  static usage = '<%= command.id %> [options]';
```
Comment (em-c-rod) [inline src/commands/emasser/delete/poams.ts:14]: What is the difference between ARGUMENTS and options?

File: src/commands/emasser/get/artifacts.ts (hunk 2 above)
Comment (georgedias) [reply src/commands/emasser/get/artifacts.ts:16]: bin/run emasser get artifacts -h or --help

File: src/commands/emasser/delete/poams.ts (hunk 3 above)
Comment (georgedias) [reply src/commands/emasser/delete/poams.ts:14]: options are actually the FLAGS, I contemplated what to put there.
I think I should just change all to [FLAGS] - what do you think?

**Review by em-c-rod — APPROVED** [review]
LGTM

**Pull request merged**